libcaf_core/caf/detail/json.cpp

```diff
@@ -312,7 +413,9 @@ void read_json_string(ParserState& ps, std::vector<char>& scratch_space,
     transition(read_chars, 'r', scratch_space.push_back('\r'))
     transition(read_chars, 't', scratch_space.push_back('\t'))
     transition(read_chars, 'v', scratch_space.push_back('\v'))
+    fsm_transition(read_4_hex(ps, append_char), read_chars, 'u', ( has_escaped()))
```

Comment:
I've taught this through as well. The question is do we really want the whole json parsing to fail if one code point is malformed. Leaving the content of the string as raw `\uXXXX` instead of transforming to utf8 seems more relaxed in this case?

Reply:
Hm, I do see your point. Mind looking into what other libraries like RapidJSON, Qt, or nlohman do in this case?

Reply:
Done. `nlohmann` and `RapidJSON`  both fail to parse when the surrogate pair is missing. Furthermore, `nlohmann` also throws an exception when the 4 hex sequence is invalid, while RapidJson accepts it with some invalid bytes in the resulting string. 
I'll refactor this, in fact I have something similar in the commit history.